cli/azd/pkg/commands/builder.go

```diff
@@ -16,18 +17,18 @@ func Build(action Action, rootOptions *GlobalCommandOptions, use string, short s
 		Long:  long,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			ctx := context.Background()
-			azdCtx, err := environment.NewAzdContext()
+			azdCtx, err := azd_context.NewAzdContext()
 			if err != nil {
 				return fmt.Errorf("creating context: %w", err)
 			}
 
 			// Set the global options in the go context
-			ctx = context.WithValue(ctx, environment.AzdContextKey, azdCtx)
-			ctx = context.WithValue(ctx, environment.OptionsContextKey, rootOptions)
+			ctx = azd_context.WithAzdContext(ctx, azdCtx)
```

Comment:
If we're going to store the `AzdContext` in the context, then I think we should avoid passing around the `azdCtx` variable so much through the rest of the system.

Reply:
Yes, I can create a subsequent PR for that.

I would like us to start creating patterns and coding guidelines around our components.